File: packages/javascript/src/demo-uncovered.ts
/**
 * DEMONSTRATION ONLY — do not merge.
 *
 * New `src/` code that is exercised purely by an integration test
 * (`tests/demo-uncovered.test.ts`) and has NO colocated unit test
 * (`src/demo-uncovered.test.ts`). Used to confirm the coverage gate
 * fails when `src/` lacks unit coverage.
 */
export const demoUncovered = (a: number, b: number): number => {
  if (a > b) {
    return a - b;
  }
  return a + b;
};

File: packages/python/src/cachetta/demo_uncovered.py
"""DEMONSTRATION ONLY — do not merge.

New ``src/`` code exercised purely by an integration test
(``tests/demo_uncovered_test.py``, marked ``integration``) with NO
colocated unit test. Used to confirm the coverage gate fails when
``src/`` lacks unit coverage and integration tests are excluded.
"""


def demo_uncovered(a: int, b: int) -> int:
    """Return ``a - b`` when ``a > b`` else ``a + b``."""
    if a > b:
        return a - b
    return a + b
